app/client/src/style/OlStyleDefs.js

```diff
@@ -1031,19 +1032,33 @@ export function ptStationCountStyle(feature, resolution) {
       })
     });
   } else {
-    const colorMapping = {
-      0: "#D31E20",
-      1: "#004D89",
-      2: "#338FB4",
-      3: "#005567",
-      4: "#BEB8EB",
-      5: "#1A3A3A",
-      6: "#C57B57",
-      7: "#457B9D",
-      11: "#6A3E37",
-      12: "#34344A"
-    };
-    const colors = Object.keys(tripCnt).map(key => colorMapping[key]);
+    const transportTypes = appStore.state.appConfig.routing[3].transit_modes;
+
+    const result = transportTypes.map(transport =>
+      publicTransportStations(transport.icon, transport.color)
+    );
+
```

Comment:
It will be good to have this as a getter or hardcoded when app is loaded, as we have to loop 4-5 times on every feature request. .